pkg/sentry/fsimpl/fuse/connection.go

```diff
@@ -54,24 +76,258 @@ type FutureResponse struct {
 // Connection is the struct by which the sentry communicates with the FUSE server daemon.
 type Connection struct {
 	fd *DeviceFD
+
+	// MaxRead size in bytes
+	MaxRead uint32
+
+	// MaxWrite size in bytes
+	MaxWrite uint32
+
+	// MaxPages is the maximum number of pages for a single request to use
+	MaxPages uint16
+
+	// MaxBackground is the maximum number of outstanding background requests
```

Comment:
I'm a little confused to what a background request is in context of FUSE.  I ask because I don't see variables for `MaxRequests` in general, just background - what are they and why do we need to have limits for them and not all requests?

Reply:
Based on my limited understanding on the unix/linux code, there are 3 types of requests:

1. requests the users initiated and to be sent
2. requests sent and waiting for reply
3. requests with reply and to be processed

For the 3 above, they also need to be divided into **subgroups** based on if the device buffer is full and blocked on writing and so on. So it's quite complicated and I don't have a crystal clear picture yet